cdb2api/cdb2api.c

```diff
@@ -2497,14 +2500,14 @@ static int cdb2_random_int()
     return nrand48(rand_state);
 }
 
-/* Get random value from range 0 to max-1 excluding 1 value */
-static int getRandomExclude(int max, int exclude)
+/* Get random value from range min to max-1 excluding 1 value */
+static int getRandomExclude(int min, int max, int exclude)
 {
     int val = 0;
-    if (max < 2)
+    if (max < 2 || max <= min)
```

Comment:
In the previous code, if there's only 1 possible value, the code would always return that value (which would have been 0), whatever the excluded value happens to be.  So maybe equivalent is 'if (max - min < 2) return min'

Reply:
Good catch! Thank you!